aicsimageio/readers/lif_reader.py

```diff
@@ -0,0 +1,767 @@
+        read_length: numpy.ndarray
+            A 1D numpy array of read lengths, the index is the scene index
+        read_dims: Optional[Dict[str, int]]
+            The dimensions to read from the file as a dictionary of string to integer.
+            Default: None (Read all data from the image)
+
+        Returns
+        -------
+        data: np.ndarray
+            The data read for the dimensions provided.
+        read_dimensions: List[Tuple[str, int]]]
+            The dimension sizes that were returned from the read.
+        """
+        # Catch optional read dim
+        if read_dims is None:
+            read_dims = {}
+
+        # Read image
+        log.debug(f"Reading dimensions: {read_dims}")
+        data, dims = LifReader._get_item_as_bitmap(img, offsets, read_length, meta, read_dims)
+
+        # Drop dims so that the data dims match the chunk_dims for dask
+        ops = []
+        real_dims = []
+        for i, dim_info in enumerate(dims):
+            # Expand dimension info
+            dim, size = dim_info
+
+            # If the dim was provided in the read dims we know a single plane for that
+            # dimension was requested so remove it
+            if dim in read_dims:
+                ops.append(0)
+            # Otherwise just read the full slice
+            else:
+                ops.append(slice(None, None, None))
+                real_dims.append(dim_info)
+
+        # Convert ops and run getitem
+        return data[tuple(ops)], real_dims
+
+    @staticmethod
+    def _imread(img: Path, offsets: List[np.ndarray],
+                read_length: np.ndarray,
+                meta: Element,
+                read_dims: Optional[Dict[str, str]] = None
+                ) -> np.ndarray:
+        """
+        This function is a pass through to _read_image above
+        the difference is it returns the data without the dims structure.
+        Parameters
+        ----------
+        img: Path
+            Path to the LIF file to read.
+        offsets: List[numpy.ndarray]
+            A List of numpy ndarrays offsets, see _compute_offsets for more details.
+        read_length: numpy.ndarray
+            A 1D numpy array of read lengths, the index is the scene index
+        read_dims: Optional[Dict[str, int]]
+            The dimensions to read from the file as a dictionary of string to integer.
+            Default: None (Read all data from the image)
+
+        Returns
+        -------
+        data: np.ndarray
+            The data read for the dimensions provided.
+        """
+        data, dims = LifReader._read_image(img=img,
+                                           offsets=offsets,
+                                           read_length=read_length,
+                                           meta=meta,
+                                           read_dims=read_dims
+                                           )
+        return data
+
+    @staticmethod
+    def _daread(
```

Comment:
Since this function is entirely copied from CZI Reader I will probably try to merge the two functions together at a later time. This is a note for me. No worries.

Reply:
@JacksonMaxfield  #101